### PR TITLE
Fix linkify to only link to proper URLs

### PR DESCRIPTION
**Pull request by perlpunk**
This doesn't check for really well formed URLs, but it limits the matching to only allowed characters in URIs.
It makes up for escapeForHtml() being already called, so it matches `&amp;` separately.

Tested with:

    # "> should be excluded
    "https://github.com/os-autoinst/openQA.git"
    <https://github.com/os-autoinst/os-autoinst.git>
    # ) should be included
    https://de.wikipedia.org/wiki/Queen_(Band)
    # query strings should be included
    https://github.com/os-autoinst/openQA?foo=1%20&bar=*2

Single quotes actually can be part of URIs so that's something we should adapt
in our code (and use double quotes, or I remember back in usenet times we used
`<http://...>` whenever referring to a URL in plain text).

We could use this as a workaround until an according PR to https://github.com/IonicaBizau/anser has been made.
There is a related issue: https://github.com/IonicaBizau/anser/issues/63

Alternative to
* https://github.com/os-autoinst/os-autoinst/pull/2544

Issue: https://progress.opensuse.org/issues/166676

### CODE REVIEW

**Review by okurz — CHANGES_REQUESTED** [review]

**Review by okurz — COMMENTED** [review]
+1

File: assets/javascripts/anser-import.js [hunk 1]
```diff
@@ -1,7 +1,13 @@
 const module = {};
 
+function linkify(txt) {
+  const re = /(https?:\/\/([A-Za-z0-9#;/?:@=+$,_.!~*()[\]-]|&amp;)+)/gm;
```
[the remark this answers is not in the document]
Comment (okurz) [reply assets/javascripts/anser-import.js]: I would be ok if we change all use of `'` to `"` when quoting URLs

**Comment by perlpunk** [comment]
https://github.com/IonicaBizau/anser/pull/75

**Comment by perlpunk** [comment]
I added a test

**Review by Martchus — APPROVED** [review]

**Review by b10n1k — APPROVED** [review]

**Pull request merged**